shared/src/vyxal/NumberHelpers.scala

```diff
@@ -257,6 +257,12 @@ object NumberHelpers:
     val step = if start < end then 1 else -1
     start.to(end, step = step)
 
+  def range(start: VNum, ends: Seq[VNum])(using Context): VList =
+    val ranges = ends.map(start.to(_))
+    ListHelpers
+      .reshape(ListHelpers.cartesianProductMulti(ranges), ranges.map(_.length))
+      .asInstanceOf[VList]
```

Comment:
Is `.asInstanceOf` safe to use here? You could either make `reshape`'s return type `VList` or wrap this in `VList.from`:
```suggestion
    VList.from(
      ListHelpers
        .reshape(ListHelpers.cartesianProductMulti(ranges), ranges.map(_.length))
    )
```

Reply:
Reshape returns a scalar (not a `VList`) if the shape is an empty list. I suppose this might not be safe if range is called with an empty list but that could just be an error (though I guess a more clear error might be useful)

Reply:
alright, I added an error message, now `.asInstanceOf[VList]` is safe